apps/backend/src/services/zfs/analysis_service.py

```
@@ -0,0 +1,173 @@
+            # Group by dataset
+            for snapshot in snapshots:
+                dataset_name = snapshot["name"].split("@")[0]
+                if dataset_name not in usage_analysis["snapshots_by_dataset"]:
+                    usage_analysis["snapshots_by_dataset"][dataset_name] = []
+                usage_analysis["snapshots_by_dataset"][dataset_name].append(snapshot)
+            
+            # Find largest snapshots (sort by used space if numeric)
+            try:
+                sorted_snapshots = sorted(snapshots, key=lambda x: self._parse_size(x["used"]), reverse=True)
+                usage_analysis["largest_snapshots"] = sorted_snapshots[:10]
+            except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)